openedx/features/wikimedia_features/meta_translations/api/v0/utils.py

```diff
@@ -0,0 +1,220 @@
+        data_map = { random_key: data }
+        base_data_map = { random_key: base_data }
+        return data_map, base_data_map
+    
+    random_key = get_random_string()
+    data, base_data = get_block_data_from_table(block, wiki_objects)
+    data_map = { random_key: data }
+    base_data_map = { random_key: base_data }
+
+    data_map[random_key]['children'] = {}
+    base_data_map[random_key]['children'] = {}
+
+    for child in block.get_children():
+        course_outline, course_base_outline = get_recursive_blocks_data_from_table(child, wiki_objects, depth - 1)
+        data_map[random_key]['children'].update(course_outline)
+        base_data_map[random_key]['children'].update(course_base_outline)
+    return data_map, base_data_map
+
+def get_course_outline(course_id, N=3):
+    """
+    Get course outline of a course
+    Arguments:
+        course_id: Course ID
+        N: depth
+    Return:
+        dict: structured outline of a course based on depth
+    """
+    wiki_objects = WikiTranslation.objects.filter(target_block__course_id=course_id)
+    course_key = CourseKey.from_string(course_id)
+    course = get_course_by_id(course_key)
+    course_data, base_course_data = get_recursive_blocks_data_from_table(course, wiki_objects, N)
+    return course_data, base_course_data
+
+def get_outline_course_to_units(course):
+    """
+    Get course and base course outline from course to units
+    Arguments:
+        course: Course Block
+    Returns:
+        dict: course-outline
+        dict: base course-outline
+    """
+    wiki_objects = WikiTranslation.objects.filter(target_block__course_id=course.course_id)
+    return get_recursive_blocks_data_from_table(course, wiki_objects, 3)
+
+def get_outline_unit_to_components(unit):
+    """
+    Get course and base course outline from units to component
+    Arguments:
+        course: Vertical Course Block - Unit
+    Returns:
+        dict: course-outline
+        dict: base course-outline
+    """
+    wiki_objects = WikiTranslation.objects.filter(target_block__course_id=unit.course_id)
+    return get_recursive_blocks_data_from_table(unit, wiki_objects)
+
+def get_course_version_object(course_key):
+    """
+    Arguments:
+        course_key: Course Key
+    Retruns:
+        version obj: dic(id, title, language)
+    """
+    course = get_course_by_id(course_key)
+    version_obj = {
+        'id': str(course.course_id),
+        'title': course.display_name,
+        'language': course.language
+        }
+    return str(course.course_id), version_obj
+
+def get_courses_of_base_course(base_course_id):
+    """
+    Returns:
+        Translated versions of a base course
+    """
+    course_varsions = CourseTranslation.objects.filter(base_course_id = base_course_id)
+    translated_courses = [get_course_version_object(course_version.course_id) for course_version in course_varsions]
+    return dict(translated_courses)
```

Comment:
The format required by Rahat bahi was like this. It becomes easier for him to insert JSON objects into react. With your recommendation we could sent list object instead of json one. 
```
 {
    course_key_2: {
        "id": "course_key_2",
        "tilte": "Introduction To Computing(Urdu)",
        "language": "ur",
    },
    course_key_3: {
        "id": "course_key_3",
        "tilte": "Introduction To Computing(French)",
        "language": "fr",
    }
}
```